docs/data-guidelines.md

```diff
@@ -12,6 +12,45 @@ If it's helpful to understanding the rule, summarize the rationale. Definitely c
 
 -- END TEMPLATE -->
 
+
+## Adding features and subfeatures
```

Comment:
@queengooborg Updated as per our discussion:

- features required for all released web platform features. Term feature is now defined.
- subfeatures term now defined
- subfeature required for interface methods and properties by default if there is any released implementation.
- subfeatures not requried for anything else unless there is a change in version support. This has nothing to do with what the spec is doing and is entirely due to some implementation change.

I think this is at least close, though there might be other exceptions like interface methods and properties. See comment below.

There is also the slight problem of things like HTTP headers which are not necessarily implemented in a browser, and for which the term "supported" is very loose. I think we should get this in and then address that separately.